### PR TITLE
LUKS metadata size

### PR DESCRIPTION
**Pull request by vojtechtrefny**
New version of #530 with locks (should prevent crashes on CentOS 7).

### CODE REVIEW

**Comment by vojtechtrefny** [comment]
> I think the locking should be done in libblockdev because it is supposed to be thread-safe.

It is possible to run multiple crypto functions on different devices and it would be really hard to implement "per device" locks in libblockdev.

**Comment by vpodzime** [comment]
> it would be really hard to implement "per device" locks in libblockdev.

It wouldn't be that hard, but since this is actually an issue only in older versions of *libgcrypt* (used by *libcryptsetup*), it's probably not worth it. Let's just deal with this in *UDisks* then.

File: src/udiskslinuxblock.c [hunk 1]
```diff
@@ -2727,6 +2746,20 @@ udisks_linux_block_is_unknown_crypto (UDisksBlock *block)
 
 /* ---------------------------------------------------------------------------------------------------- */
 
+void
+udisks_linux_block_set_encrypted_lock (UDisksBlock *block,
+                                       gboolean    lock)
```
Comment (vpodzime) [inline src/udiskslinuxblock.c]: I think it would be a lot nicer to have `lock/unlock` or `acquire/release` methods.

Comment (vojtechtrefny) [reply src/udiskslinuxblock.c]: Ok, fixed.

**Pull request merged**